Sources/AsyncHTTPClient/Utils.swift

```diff
@@ -77,149 +78,207 @@ extension NIOClientTCPBootstrap {
+                        } catch {
+                            return channel.eventLoop.makeFailedFuture(error)
+                        }
                     }
-                    return channel.eventLoop.makeSucceededVoidFuture()
-                } catch {
-                    return channel.eventLoop.makeFailedFuture(error)
-                }
             }
     }
 
-    static func makeHTTP1Channel(destination: ConnectionPool.Key, eventLoop: EventLoop, configuration: HTTPClient.Configuration, preference: HTTPClient.EventLoopPreference) -> EventLoopFuture<Channel> {
+    static func makeHTTP1Channel(destination: ConnectionPool.Key,
+                                 eventLoop: EventLoop,
+                                 configuration: HTTPClient.Configuration,
+                                 sslContextCache: SSLContextCache,
+                                 preference: HTTPClient.EventLoopPreference,
+                                 logger: Logger) -> EventLoopFuture<Channel> {
         let channelEventLoop = preference.bestEventLoop ?? eventLoop
 
         let key = destination
-
         let requiresTLS = key.scheme.requiresTLS
-        let bootstrap: NIOClientTCPBootstrap
-        do {
-            bootstrap = try NIOClientTCPBootstrap.makeHTTPClientBootstrapBase(on: channelEventLoop, host: key.host, port: key.port, requiresTLS: requiresTLS, configuration: configuration)
-        } catch {
-            return channelEventLoop.makeFailedFuture(error)
+        let sslContext: EventLoopFuture<NIOSSLContext?>
+        if key.scheme.requiresTLS {
+            sslContext = sslContextCache.sslContext(tlsConfiguration: configuration.tlsConfiguration ?? .forClient(),
+                                                    eventLoop: eventLoop,
+                                                    logger: logger).map { $0 }
+        } else {
+            sslContext = eventLoop.makeSucceededFuture(nil)
         }
 
-        let channel: EventLoopFuture<Channel>
-        switch key.scheme {
-        case .http, .https:
-            let address = HTTPClient.resolveAddress(host: key.host, port: key.port, proxy: configuration.proxy)
-            channel = bootstrap.connect(host: address.host, port: address.port)
-        case .unix, .http_unix, .https_unix:
-            channel = bootstrap.connect(unixDomainSocketPath: key.unixPath)
-        }
+        let bootstrap = NIOClientTCPBootstrap.makeHTTPClientBootstrapBase(on: channelEventLoop,
+                                                                          host: key.host,
+                                                                          port: key.port,
+                                                                          requiresTLS: requiresTLS,
+                                                                          configuration: configuration,
+                                                                          sslContextCache: sslContextCache,
```

Comment:
Why do we pass the `SSLContextCache` down here? We already have a `sslContext` (a `Future`) created before...

Reply:
That is an excellent question, I have no idea. I'll look into this, maybe we can just inject the SSLContext into that method.